swap/swap.go

```diff
@@ -405,17 +403,18 @@ func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitCheque
 
 	otherSwap, err := contract.InstanceAt(cheque.Contract, s.backend)
 	if err != nil {
-		log.Error("error getting contract", "err", err)
-		return err
+		return fmt.Errorf("error getting contract: %w", err)
 	}
 
 	gasPrice, err := s.backend.SuggestGasPrice(context.TODO())
 	if err != nil {
-		return err
+		// todo: handle nil error
```

Comment:
It is better to log the error here if it is not propagated.

Reply:
I agree. The idea was to introduce a special error that is used when the loop is to be finished, and to just return error in the cases like this so it can be logged by `protocol.Run()`. Do you think it makes sense or should we just log in the handlers and return nil?

Reply:
Basically the usage difference would be:
1. In order to log an error, but not break the loop, return err
2. In ordeer to log and error and break the loop, use special error

Or we could just log in the handler, then the usage would be:
1. In order to log an error and break the loop -  return err
2. In order to log and error but not break the loop - log err in handler and return nil

Reply:
It makes sense. If you would like to leave new error type for another PR, then it is ok to just log the error here (in this PR), as this is its last handling, for now.

Reply:
Sounds good, thanks!

Reply:
It would be good to hear @zelig's opinion also. I think that DropError type which encapsulates original error would make sense, as we could have also BlacklistError in a similar manner, but would contain additional information about blacklisting period or other information.

Reply:
Makes sense to me as well. I will just log an error here then at this point, and will open a separate PR for `DropError` then, and we can discuss there.

Reply:
I ended up doing it in the same PR, as it is part of the similar functionality, and I wanted to avoid you guys needing to go through very similar code and PRs twice, especially chronologically related. I hope that is good :)